mqtt/mqtt-bridge/src/client.rs

```diff
@@ -158,8 +280,10 @@ impl<T: EventHandler> MqttClient<T> {
     pub async fn handle_events(mut self) -> Result<(), ClientConnectError> {
         while let Some(event) = self.client.try_next().await.unwrap_or_else(|e| {
             error!(message = "failed to poll events", error=%e);
+            // TODO: handle the error by recreting the connection
             None
         }) {
+            debug!("handle event {:?}", event);
```

Comment:
Maybe just bubble the error up and log it there? or, alternatively, you can use `loop { match ... }`
```suggestion
while let Some(event) = self.client.try_next().await? {
```

Reply:
this can't be bubbled up because we don't want to end the thread, the missing part is handling this error properly and recreating the client. The issue will be solved by the pump implementation.